app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt

```diff
@@ -437,13 +422,7 @@ class ThreadFragment : Fragment() {
         if (hasUsableCache(requireContext()) || isInlineCachedFile(requireContext())) {
             startActivity(openWithIntent(requireContext()))
         } else {
-            safeNavigate(
-                ThreadFragmentDirections.actionThreadFragmentToDownloadAttachmentProgressDialog(
-                    attachmentResource = resource!!,
-                    attachmentName = name,
-                    attachmentType = getFileTypeFromMimeType(),
-                )
-            )
+            mainViewModel.downloadAttachmentsArgs.value = Triple(resource!!, name, getFileTypeFromMimeType())
```

Comment:
Does it make sens to turn this into a `SingleLiveEvent` rather than defining a method to call from here that directly executes what needs to be executed? Because it's only triggered from here and is only called once because it's a `SingleLiveEvent`, no?

Reply:
We need to think about it but maybe it could even make sens to extract it to a separate use case file depending on the situation

Reply:
We need to navigate, and since we are in the ThreadFragment, we are out of navigation. So we can't do it here.

Reply:
We can extract the `//region Current Thread` to a TwoPaneViewModel, to help lighten the MainViewModel.

Reply:
Done here https://github.com/Infomaniak/android-kMail/pull/1606?w=1